analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -267,11 +271,20 @@ open class Npm(
 
         log.info { "Searching for 'package.json' files in '$nodeModulesDir'..." }
 
-        nodeModulesDir.walk().filter {
+        val nodeModulesFiles = nodeModulesDir.walk().filter {
             it.name == "package.json" && isValidNodeModulesDirectory(nodeModulesDir, nodeModulesDirForPackageJson(it))
-        }.forEach { file ->
-            val (id, pkg) = parsePackage(rootDirectory, file)
-            packages[id] = pkg
+        }
+
+        runBlocking(Dispatchers.IO) {
+            nodeModulesFiles.map { file ->
+                this@Npm.log.debug { "Starting to parse '$file'..." }
+                async { file to parsePackage(rootDirectory, file) }
+            }.forEach { deferredResult ->
+                val (file, result) = deferredResult.await()
+                val (id, pkg) = result
+                this@Npm.log.debug { "Finished parsing '$file' to '$id'." }
+                packages[id] = pkg
+            }
```

Comment:
This could be simplified a bit:

```kotlin
        return runBlocking {
            nodeModulesDir.walk().filter {
                it.name == "package.json" && isValidNodeModulesDirectory(nodeModulesDir, nodeModulesDirForPackageJson(it))
            }.map { file ->
                async {
                    this@Npm.log.debug { "Starting to parse '$file'..." }
                    parsePackage(rootDirectory, file).also { (id, _) ->
                        this@Npm.log.debug { "Finished parsing '$file' to '$id'." }
                    }
                }
            }.toList().awaitAll().toMap()
        }
```

Reply:
I basically had it like that before, but was bewildered by not seeing any concurrency at all according to log output, so I wanted to put the "Finished" log after the `await()` for the respective file to see if that makes a change. But it didn't 😟